compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -54,10 +60,24 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
         .filter(_.nn.getName.nn.startsWith("scoverage"))
         .foreach(_.nn.delete())
     end if
+
+    coverageExcludeClasslikePatterns = ctx.settings.coverageExcludeClasslikes.value.map(_.r.pattern)
+    coverageExcludeFilePatterns = ctx.settings.coverageExcludeFiles.value.map(_.r.pattern)
+
     super.run
 
     Serializer.serialize(coverage, outputPath, ctx.settings.sourceroot.value)
 
+  private def isClassIncluded(sym: Symbol)(using Context): Boolean =
+    coverageExcludeClasslikePatterns.isEmpty || !coverageExcludeClasslikePatterns.exists(
+      _.matcher(sym.fullName.toText(ctx.printerFn(ctx)).show).nn.matches
+    )
+
+  private def isFileIncluded(file: SourceFile)(using Context): Boolean =
+    coverageExcludeFilePatterns.isEmpty || !coverageExcludeFilePatterns.exists(
+      _.matcher(file.path.replace(".scala", "")).nn.matches
```

Comment:
Nit: extract `file.path.replace(".scala", "")` in a `val` outside of the `exists`?